From d6f8723b8caa97a355ef7311dae3b8d630a2eb12 Mon Sep 17 00:00:00 2001
From: Alexander Akulich <akulichalexander@gmail.com>
Date: Thu, 22 Oct 2020 01:41:52 +0300
Subject: [PATCH] [qmimodem] Fix dialing to an international number

We can't just use ph->number because it has no leading '+' which is
crucial for a call (the operator says "The called number is not valid"
in case of a number without the plus).

Use phone_number_to_string() to get the valid number.
---
 drivers/qmimodem/voicecall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/qmimodem/voicecall.c b/drivers/qmimodem/voicecall.c
index cfc6f0b..e848c98 100644
--- a/drivers/qmimodem/voicecall.c
+++ b/drivers/qmimodem/voicecall.c
@@ -241,7 +241,7 @@ static void dial(struct ofono_voicecall *vc, const struct ofono_phone_number *ph
 
 	cbd->user = vc;
 	arg.calling_number_set = true;
-	arg.calling_number = ph->number;
+	arg.calling_number = phone_number_to_string(ph);
 	memcpy(&vd->dialed, ph, sizeof(*ph));
 
 	arg.call_type_set = true;
-- 
2.26.2

